src/transformers/pytorch_utils.py

```diff
@@ -326,3 +331,24 @@ def isin_mps_friendly(elements: torch.Tensor, test_elements: torch.Tensor | int)
     else:
         # Note: don't use named arguments in `torch.isin`, see https://github.com/pytorch/pytorch/issues/126045
         return torch.isin(elements, test_elements)
+
+
+def translate_to_torch_parallel_style(style: str):
+    """
+    In model configurations, we use a neutral type (string) to specify parallel
+    styles, here we translate them into torch.distributed tensor-parallel
+    types.
+    """
+    if not isinstance(style, str):
+        raise ValueError(
+            f"Unsupported parallel style type {type(style)}, expected str"
+        )
+
+    if style == "colwise":
+        return ColwiseParallel()
+    elif style == "rowwise":
+        return RowwiseParallel()
+    elif style == "colwise_rep":
+        return ColwiseParallel(output_layouts=Replicate())
+    else:
+        raise ValueError(f"Unsupported parallel style value: {style}")
```

Comment:
A mapping from tp style to the correct function might be better. 
We are also thinking of defining a TensorParallelConfig, your feedback is welcome here, as I don't know the variety of classes / args that might be used!

Reply:
Thanks for the comment! Indeed a mapping style would look better. 
The only caveat is that the returned value here is an object rather than a constant or class (see the `()` behind `ColwiseParallel`). If we prepare a map, we may be always returning the *same* object -- the `parallelize_module` API may be able to support it I guess, I am just not sure if there is a contract guaranteeing that today.

Reply:
Pitching in :)

We should be able to use the same object since it applies required parallel operation to the module and returns a new copy - https://github.com/pytorch/pytorch/blob/86d4b7d60b264cae5a04a1b20719bcd7a5752a4c/torch/distributed/tensor/parallel/api.py#L95 

Have also tested it empirically while benchmarking (https://github.com/huggingface/transformers/pull/34194)

Thanks!

Reply:
SOunds good!